### ISSUE TITLE
Alsamixer not detecting headphones

### ISSUE THREAD

**Issue by lgmventura** [alsa-project/alsa-ucm-conf#283]
I'm using the motherboard ASUS ROG Strix B760-f Gaming Wifi, which has 7.1 audio channels on the back side and connectors for an audio panel at the front of the case - a microphone and a headphones jack.

When I plug my headphones into the front channels connector (at the back of the PC case) there is normal sound. When I plug it into the headphones jack (at the front of the case), it gets automatically detected by `pavucontrol`, which switches to the headphones as output, as it should be:
![image](https://user-images.githubusercontent.com/16710238/221439991-903fd90b-43b2-41a0-af7f-1ed02085238e.png)

However, there is no sound. When I open `alsamixer` and choose the respective soundcard, which is weirdly called "USB audio" there, I see no volume control for the headphones, as there should be. All it shows is this:
![image](https://user-images.githubusercontent.com/16710238/221440193-381eadfb-fed9-48ad-807b-05c736b65b24.png)

The only way I got to get audio output from the headphones jack is by running `aplay` and choosing the hardware directly from there `aplay -D plughw:1,1 /usr/share/sounds/alsa/Front_Center.wav` or on Audacity by selecting "USB Audio 1: (hw:1,1)" as playback device.

I also tried to use JACK Audio Connection Kit, but found no configuration there that routes the audio to the headphones jack, even trying all 4 possible analogue outputs.

System:
- Motherboard ASUS ROG Strix B760-f Gaming Wifi
- Intel core i7-13700K
- Ubuntu 22.04 (freshly installed)

**Comment by perexg** [comment]
Attach output from `alsa-info.sh --no-upload`, please.

**Comment by perexg** [comment]
> which is weirdly called "USB audio" there

You have really an USB sound card on the motherboard with ALC4080/82 USB chip and the hardware vendor didn't bother to set a better text to the USB audio descriptor table.

**Comment by lgmventura** [comment]
Where do I find this file? I get `alsa-info.sh: command not found`

**Comment by lgmventura** [comment]
Is maybe this? `/usr/sbin/alsa-info`

**Comment by lgmventura** [comment]
I got this file: <script src="https://gist.github.com/lgmventura/df505dcf771ff349414eb8ed39ae21e4.js"></script>

**Comment by perexg** [comment]
It should be already supported - follow https://github.com/alsa-project/alsa-ucm-conf/blob/master/README.md .

**Comment by lgmventura** [comment]
Curl downloaded the file correctly, but when I run `tar xvzf alsa-ucm-conf.tar.gz -C /usr/share/alsa --strip-components=1 "*/ucm" "*/ucm2"`, I'm getting:
```
tar: Pattern matching characters used in file names
tar: Use --wildcards to enable pattern matching, or --no-wildcards to suppress this warning
tar: */ucm: Not found in archive
tar: Pattern matching characters used in file names
tar: Use --wildcards to enable pattern matching, or --no-wildcards to suppress this warning
tar: */ucm2: Not found in archive
tar: Exiting with failure status due to previous errors
```

**Comment by lgmventura** [comment]
Although the folders are there:
![image](https://user-images.githubusercontent.com/16710238/221442278-2e390371-86b2-4a43-80ea-d83a05aeb7f3.png)

**Comment by perexg** [comment]
I corrected the command for your distro. See again https://github.com/alsa-project/alsa-ucm-conf/blob/master/README.md .

**Comment by lgmventura** [comment]
I managed to run the 2 lines following "The latest configuration can be obtained with those commands:". The second one only with sudo. I rebooted the system and alsamixer still looks exactly the same as before. Am I missing something?

**Comment by perexg** [comment]
Which version of the alsa-lib package do you have in your distro ? Do you see an error with the command `alsaucm -c 0 dump text` ?

**Comment by lgmventura** [comment]
Hi, checking `cat /proc/asound/version`, I get: `Advanced Linux Sound Architecture Driver Version k5.19.0-32-generic.`.
The command `alsaucm -c 0 dump text` gives me:
```
ALSA lib main.c:1412:(snd_use_case_mgr_open) error: failed to import 0 use case configuration -2
alsaucm: error failed to open sound card 0: No such file or directory
```
I also tested with `-c 1`, but it gives me the same error with value 1 instead.

**Comment by perexg** [comment]
Sorry, it should be `alsaucm -c hw:0 dump text`.

**Comment by lgmventura** [comment]
Thanks! I get this output:
```
ALSA lib parser.c:2021:(parse_master_file) Incompatible syntax 6 in HDA-Intel.conf
ALSA lib main.c:1412:(snd_use_case_mgr_open) error: failed to import hw:0 use case configuration -22
alsaucm: error failed to open sound card hw:0: Invalid argument
```

**Comment by perexg** [comment]
Then try `hw:1`. 0 is probably your HDMI sound card.

**Comment by lgmventura** [comment]
The same:
```
ALSA lib parser.c:2021:(parse_master_file) Incompatible syntax 6 in USB-Audio.conf
ALSA lib main.c:1412:(snd_use_case_mgr_open) error: failed to import hw:1 use case configuration -22
alsaucm: error failed to open sound card hw:1: Invalid argument
```
But for 2, it's slightly different:
```
ALSA lib confmisc.c:165:(snd_config_get_card) Cannot get card index for 2
ALSA lib parser.c:2412:(uc_mgr_import_master_config) card 'hw:2' is not valid
ALSA lib main.c:1412:(snd_use_case_mgr_open) error: failed to import hw:2 use case configuration -2
alsaucm: error failed to open sound card hw:2: No such file or directory
```

**Comment by perexg** [comment]
It's not same:

```
ALSA lib parser.c:2021:(parse_master_file) Incompatible syntax 6 in USB-Audio.conf
```

Upgrade alsa-lib...

**Comment by lgmventura** [comment]
Ah, yes, true. How can I upgrade it on Ubuntu? I tried with:
```
sudo apt update
sudo apt install libasound2
```
according to an online help and I get: `libasound2 is already the newest version (1.2.6.1-1ubuntu1).`.

**Comment by perexg** [comment]
Unfortunately, you are asking the upstream for the support not your distribution. The alsa-lib 1.2.8 was released before 4 months, 1.2.7 before 7 months. Move your question to your distro support.

**Comment by lgmventura** [comment]
Sorry. I'll check there, thanks. What distro do you use? Do you know of a distro that is more friendly to the newest versions of packages?

**Comment by lgmventura** [comment]
Hi, I just tested it on Ubuntu 22.10 (preview mode, i.e. without installing it). This should have alsa-lib 1.2.7.2 by default (see https://launchpad.net/ubuntu/+source/alsa-lib). The appearance of alsamixer is the same (not showing headphones) and those commands produce:
```
$ alsaucm -c hw:0 dump text
ALSA lib parser.c:244:(error_node) UCM is not supported for this HDA model (HDA Intel PCH at 0x6003220000 irq 196)
ALSA lib main.c:1541:(snd_use_case_mgr_open) error: failed to import hw:0 use case configuration -6
alsaucm: error failed to open sound card hw:0: No such device or address
```
and
```
$ alsaucm -c hw:1 dump text
ALSA lib parser.c:244:(error_node) UCM is not supported for this USB device (Generic USB Audio at usb-0000:00:14.0-11, high speed @ USB0b05:1a52)
ALSA lib main.c:1541:(snd_use_case_mgr_open) error: failed to import hw:1 use case configuration -6
alsaucm: error failed to open sound card hw:1: No such device or address
```

**Comment by perexg** [comment]
Update to latest configuration files (tar command) like I suggested before. Those changes are not released yet.

**Comment by perexg** [comment]
Reboot and use pulseaudio / pipewire server. There is no difference in the alsamixer - UCM is a higher level.

**Comment by lgmventura** [comment]
So alsamixer will be replaced?

**Comment by perexg** [comment]
The sound server is usually used in the GUI now. If you use pavucontrol, you should see all inputs/outputs now with streams and assigned volume control. The alsamixer is the direct ALSA mixer application which can control the "bottom" things (volume controls) directly.

For USB devices, the mixer controls are described by USB descriptors. If the hardware vendor does not describe those controls in detail, you may see "PCM 1", "PCM 2" controls there. But it's hidden thing in pavucontrol (sound server) when the UCM is used.

**Comment by lgmventura** [comment]
I tested here in a new installation of Ubuntu 22.10 and it worked!!! Since there is already the `alsa-lib` 1.2.7, it was possible to run [this](https://github.com/alsa-project/alsa-ucm-conf/blob/master/README.md) that you sent and, after restarting the PC, the front headphones worked normally. But indeed alsamixer doesn't show it. But it appears on `gnome-control-center` as _front headphones_, as it should be.

**Comment by lgmventura** [comment]
Thank you so much! Do you know why some apps (like the newest version of MuseScore, for example) cannot change their output? These are also not listed in `pavucontrol`, for example.

**Comment by lgmventura** [comment]
I could also manage to make them work on Ubuntu 22.04 with the native alsa-lib 1.2.6. I installed Cadence JACK server and opened it. Then I activated Duplex Mode and selected as Output Device **hw:Audio, 1 [USB Audio]**. Then I hit OK and started the JACK server. On the sound server (pavucontrol or gnome-control-center or any other), I selected the JACK sink as output. Then, all the sound applications for which I don't configure this on themselves, will route to the "system default", which will go to `hw:1`.

**Comment by keithbowes** [comment]
>  When I plug it into the headphones jack (at the front of the case), it gets automatically detected by `pavucontrol`, which switches to the headphones as output, as it should be: ![image](https://user-images.githubusercontent.com/16710238/221439991-903fd90b-43b2-41a0-af7f-1ed02085238e.png)
> 

Lucky you. All I see is a PCM bar in alsamixer. I tried pulseaudio again (I've switched to ALSA-only because the distro I use says it's switching from PulseAudio to PipeWire in the next version and when I tried PipeWire in the current version of my distro on my previous computer, my speakers blew up (while I was adjusting the volume in alsamixer, no less), so I'm doing everything in my power to prevent that from happening again), and all pavucontrol shows in the "Ports" dropdown is "Analog Output", regardless of whether my headphones are connected or not. I know I shouldn't complain, as both the headphones and speakers work, but I would like separate settings for headphones and speakers, as in previous computers I've had.
